source/brailleDisplayDrivers/albatross/driver.py

```diff
@@ -0,0 +1,758 @@
+				)
+				time.sleep(RESET_SLEEP)
+			log.debug("I/O buffers reset done")
+			return True
+		# Considering situation where "albatross_read" thread is about to read
+		# but writing to display fails during it - or vice versa - AttributeError
+		# might raise.
+		except (IOError, AttributeError):
+			log.debug(
+				f"I/O buffer reset failed on port {self._currentPort}", exc_info=True
+			)
+			if self._dev.is_open:
+				self._dev.close()
+			return False
+
+	def _disableConnection(self):
+		"""Disables current connection after failure.
+		Reconnection retries are started.
+		"""
+		self.numCells = 0
+		if self._dev and self._dev.is_open:
+			try:
+				self._dev.close()
+				log.debug(f"Port {self._currentPort} closed")
+			except (AttributeError, OSError):
+				log.debug(f"Port {self._currentPort} close failed", exc_info=True)
+		if len(self._oldCells):
+			self._clearOldCells()
+		self._waitingCtrlPacket = False
+		self._partialCtrlPacket = None
+		self._initByteReceived = False
+		self._waitingSettingsByte = False
+		self._readQueue.clear()
+		self._writeQueue.clear()
+		self._tryToConnect = True
+
+	def _readHandling(self):
+		"""Manages all read operations.
+		Most of time called from albatross_read thread when there is something to
+		read in the port. See L{_threading} module ReadThread class.
+		When initial connection is established called from L{_searchPorts} function.
+		"""
+		if self._tryToConnect:
+			# Only one try to open port when called from "albatross_read" thread.
+			# This is indicated by _dev is not None.
+			# ReadThread run function calls _readHandling again if needed.
+			if self._dev:
+				if not self._openPort():
+					return
+			if not self._initConnection():
+				self._disableConnection()
+				return
+		data = self._somethingToRead()
+		if not data or not self._skipRedundantInitPackets(data):
+			return
+		self._handleReadQueue()
+
+	def _somethingToRead(self) -> Optional[bytes]:
+		"""All but connecting/reconnecting related read operations.
+		@return: on success returns data, on failure C{None}
+"""
+		try:
+			# If ClearCommError fails, in_waiting raises SerialException
+			if not self._dev.in_waiting:
+				return None
+			data = self._dev.read(self._dev.in_waiting)
+			log.debug(
+				f"Read: {data}, length {len(data)}, in_waiting {self._dev.in_waiting}"
+			)
+			return data
+		# Considering situation where "albatross_read" thread is about to read
+		# but writing to display fails during it - or vice versa - AttributeError
+		# might raise.
+		except (IOError, AttributeError):
+			self._disableConnection()
+			log.debug("Read failed, trying to reconnect", exc_info=True)
+			return None
+
+	def _skipRedundantInitPackets(self, data: bytes) -> bool:
+		"""Filters redundant init packets.
+		@param data: Bytes read from display.
+		@return: C{True} on success, C{False} on failure
+		"""
+		settingsByte = None
+		for i in data:
+			if not self._initByteReceived:

```

Comment:
It is probably worth converting this early and using the converted value

```suggestion
		for i in data:
			iAsByte = i.to_bytes(1, 'big')
			if not self._initByteReceived:
```